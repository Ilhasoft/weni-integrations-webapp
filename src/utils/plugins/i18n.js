import Vue from 'vue';
import VueI18n from 'vue-i18n';

Vue.use(VueI18n);

import en from '../../locales/en.json';
import pt_br from '../../locales/pt_br.json';
import es_es from '../../locales/es_es.json';

const languages = {
  'en-us': en,
  'pt-br': pt_br,
  es: es_es,
};

const messages = Object.assign(languages);

const i18n = new VueI18n({
  locale: 'en-us',
  fallbackLocale: 'en-us',
  messages,
});

export default i18n;
