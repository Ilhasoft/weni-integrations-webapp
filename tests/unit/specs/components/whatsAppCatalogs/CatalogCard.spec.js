import Vuex from 'vuex';
import { mount, createLocalVue } from '@vue/test-utils';
import CatalogCard from '@/components/whatsAppCatalogs/CatalogCard.vue';
import '@weni/unnnic-system';
import i18n from '@/utils/plugins/i18n';

const localVue = createLocalVue();
localVue.use(Vuex);

const mountComponent = async ({
  errorDisableCatalog = false,
  enabledCart = false,
  catalog = {},
} = {}) => {
  const state = {
    errorDisableCatalog,
  };

  const actions = {
    disableWhatsAppCloudCatalogs: jest.fn(),
  };

  const store = new Vuex.Store({
    modules: {
      WhatsAppCloud: {
        namespaced: true,
        actions,
        state,
      },
    },
  });

  const wrapper = mount(CatalogCard, {
    localVue,
    store,
    i18n,
    propsData: {
      catalog,
      enabledCart,
    },
  });

  return { wrapper, actions, state };
};

describe('components/whatsAppCatalog/CatalogCard.vue', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should be rendered properly', async () => {
    const { wrapper } = await mountComponent();
    expect(wrapper).toMatchSnapshot();
  });

  describe('catalog connect emits', () => {
    it('should emit enable on toggle catalog connect', async () => {
      const { wrapper } = await mountComponent();
      const toggle = wrapper.findComponent({ ref: 'catalogConnectSwitch' });
      toggle.vm.$emit('input', true);
      expect(wrapper.emitted().enable).toBeTruthy();
    });

    it('should emit disable on toggle catalog connect', async () => {
      const { wrapper } = await mountComponent();
      const toggle = wrapper.findComponent({ ref: 'catalogConnectSwitch' });
      toggle.vm.$emit('input', false);
      expect(wrapper.emitted().disable).toBeTruthy();
    });
  });

  describe('toggleCart emit', () => {
    it('should emit toggleCart on switch click', async () => {
      const { wrapper } = await mountComponent({ catalog: { is_connected: true } });
      const toggle = wrapper.findComponent({ ref: 'cartEnableSwitch' });
      toggle.trigger('click');
      expect(wrapper.emitted().toggleCart).toBeTruthy();
    });
  });
});
