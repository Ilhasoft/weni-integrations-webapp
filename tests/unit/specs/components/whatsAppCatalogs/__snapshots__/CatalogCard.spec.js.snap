// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/whatsAppCatalog/CatalogCard.vue should be rendered properly 1`] = `
<div class="whatsapp-catalog-card">
  <div class="whatsapp-catalog-card__wrapper">
    <div class="whatsapp-catalog-card__wrapper__header">
      <div class="whatsapp-catalog-card__wrapper__header__title"><span class="u font secondary body-lg color-neutral-darkest bold">

        </span></div>
      <div class="whatsapp-catalog-card__wrapper__header__description"><span class="u font secondary body-md color-neutral-cloudy">
          Identification
        </span> <span class="u font secondary body-md color-weni-600 bold">

        </span></div>
    </div>
    <div class="whatsapp-catalog-card__wrapper__actions">
      <div data-v-facf064c="" class="unnnic-switch">
        <!----><span data-v-ab84ef7c="" data-v-facf064c="" class="unnnic-icon unnnic-icon__size--sm unnnic-icon__line-height--sm unnnic-icon-scheme--neutral-soft unnnic-switch__icon"><svg id="default" viewBox="0 0 80 40" xmlns="http://www.w3.org/2000/svg"><style type="text/css"> rect { transition: fill .2s ease; } </style><rect x="4.20013" y="2.10046" width="71.6" height="35.8" rx="17.9" class="primary"></rect><circle id="default-circle" cx="22.5" cy="20.0005" r="12.5" fill="white"></circle></svg></span>
        <div data-v-facf064c="" class="unnnic-switch__label unnnic-switch__label__right unnnic-switch__label__small"> Inactive </div>
      </div>
      <!---->
    </div>
  </div>
</div>
`;
